### DISCUSSION TITLE
how do i add an onClick event over a theme icon in mantine?

### DISCUSSION THREAD

**Discussion by starxyn** [mantinedev/mantine#6298] [Q&A]
how do i add an onClick event handler over a theme icon in mantine? 
for example, i have a code like this:
`<ThemeIcon size='xs' variant='light'><IconUser></ThemeIcon>`
And I want to add an onClick event handler so that when I mouse over the IconUser picture I can launch a new modal.

**Comment by rtivital** [comment]
`<ThemeIcon size='xs' variant='light' onClick={() => console.log('click')}><IconUser></ThemeIcon>`

**Reply by starxyn** [reply]
Thank u! How about if I need to add button Props into the theme icon like how it works by adding into <Button>? I tried to add but it says error.

**Reply by rtivital** [reply]
https://mantine.dev/core/action-icon/

**Reply by starxyn** [reply]
? Sry dont understand

**Reply by atareversei** [reply]
It is better to use the `ActionIcon` component. [ActionIcon](https://mantine.dev/core/action-icon/)
Clickable elements are better defined as buttons or links. In this case, ActionIcon renders a button which will hold your desired icon.

```ts
<ActionIcon variant="light" size="xs" onClick={() => {/*do things*/}}>
  <IconUser />
</ActionIcon>
```
